src/languages/groovy.js

```diff
@@ -77,25 +92,35 @@ export default function(hljs) {
                     hljs.UNDERSCORE_TITLE_MODE
                 ]
             },
-            hljs.C_NUMBER_MODE,
             {
                 className: 'meta', begin: '@[A-Za-z]+'
             },
             {
-                // highlight map keys and named parameters as strings
-                className: 'string', begin: /[^\?]{0}[A-Za-z0-9_$]+ *:/
+              // highlight map keys and named parameters as attrs
+              className: 'attr', begin: IDENT_RE + '[ ]*:'
             },
             {
-                // catch middle element of the ternary operator
-                // to avoid highlight it as a label, named parameter, or map key
-                begin: /\?/, end: /\:/
+              // catch middle element of the ternary operator
+              // to avoid highlight it as a label, named parameter, or map key
+              begin: /\?/,
+              end: /:/,
+              contains: [
+                COMMENT,
+                STRING,
+                REGEXP,
+                NUMBER,
+                'self'
+              ]
             },
             {
                 // highlight labeled statements
-                className: 'symbol', begin: '^\\s*[A-Za-z0-9_$]+:',
+                className: 'symbol',
+                begin: '^[ ]*' + regex.lookahead(IDENT_RE + ':'),
```

Comment:
You've changed \s to space, is it ok?

Reply:
This is so the match starts on the current line, not many lines previous.

Reply:
Adding support for tabs.